dotCMS/src/main/java/com/dotcms/api/content/VanityUrlAPIImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.dotcms.api.content;
+
+import java.util.List;
+
+import com.dotcms.content.model.VanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.DotContentletStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.liferay.portal.model.User;
+
+/**
+ * Implementation class for the {@link VanityUrlAPI}.
+ * 
+ * @author oswaldogallango
+ *
+ */
+public class VanityUrlAPIImpl implements VanityUrlAPI{
+
+	final VanityUrlFactory vanityURLFactory = FactoryLocator.getVanityUrlFactory();
+
+	@Override
+	public void initializeActiveVanityURLsCache(User user) throws DotDataException, DotSecurityException {
+		this.getActiveVanityUrls(user);
+	}
+
+	@Override
+	public List<VanityUrl> getAllVanityUrls(User user) throws DotDataException, DotSecurityException {
+		return vanityURLFactory.getAllVanityUrls(user);
+	}
+
+	@Override
+	public VanityUrl getVanityUrlByURI(String uri, Host host, long languageId, User user, boolean live) 
```

Comment:
Good advise, Ozzy you can keep the method as a protected and call with true o false from the methods that Will recommend